job_executor/model/metadata.py

```diff
@@ -205,71 +209,129 @@ def dict(self, **kwargs) -> dict:  # pylint: disable=unused-argument
+        message = ''
+        if (
+            self.data_type != other.data_type or
+            self.format != other.format or
+            self.variable_role != other.variable_role
+        ):
+            message = (
+                f'dataType: {self.data_type} to {other.data_type},'
+                f'format: {self.format} to {other.format},'
+                f'variable_role: {self.variable_role} to '
+                f'{other.variable_role}\n'
+            )
+        if (
+            with_name and
+            self.name != other.name
+        ):
+            message += f'shortName: {self.name} to {other.name}\n'
+        if (
+            with_key_type and
+            self.key_type.name != other.key_type.name
+        ):
+            message += f'unitType.name: {self.key_type.name} ' \
+                       f'to {other.key_type.name}'
+
+        if message:
+            raise PatchingError(caption + message)
+
+    def validate_represented_variables(self, other: 'Variable'):
+        if len(self.represented_variables) != len(other.represented_variables):
+            raise PatchingError(
+                'Can not change the number of represented variables '
+                f'from {len(self.represented_variables)} to {len(other.represented_variables)} '
+                'Please check valueDomain.codeList field.'
+            )
+
 
 class IdentifierVariable(Variable):
-    ...
+    def patch(self, other: 'Variable') -> 'Variable':
+        if other is None:
+            raise PatchingError('Can not delete Variable')
+        # Centralized variable definition was used,
+        # don't patch the one that is in the datastore.
+        # The definition might have changed and we don't want to update it
+        # - one needs to use CHANGE operation for that.
+        return self
 
 
 class MeasureVariable(Variable):
-    ...
+    def patch(self, other: 'Variable') -> 'Variable':
+        skip_patching_represented_variables = False
+        if other is None:
+            raise PatchingError('Can not delete Variable')
+        if self.key_type is not None:
+            # Centralized variable definition was used,
+            # it is safe to only patch label and description fields.
+            self.validate_patching_fields(other, with_key_type=True)
+            skip_patching_represented_variables = True
+        else:
+            if other.key_type is not None:
+                raise PatchingError('Can not change unitType')
+            self.validate_patching_fields(other, with_name=True)
+            self.validate_represented_variables(other)
+
+        if self.not_pseudonym != other.not_pseudonym:
+            raise PatchingError(
+                'Can not change unitType''s pseudonym status from '
+                f'"{self.not_pseudonym}" to "{other.not_pseudonym}"'
+                'Please check unitType.requiresPseudonymization field. '
+            )
+
+        return super()._patch_downstream(
```

Comment:
Do not need to call super specifically here as this has also inherited the method